acra/security-controls/sql-firewall/table_filter.md

```diff
@@ -0,0 +1,29 @@
+---
+title: "Tables filters"
+bookCollapseSection: true
+---
+
+## Tables
+
+Tables work in the following manner: the names of the tables are alternately extracted from the queries that address the database and each table is compared to the list of tables from the configuration file.
+There is a restriction for query types: only `SELECT` and `INSERT` types are allowed, other query types are not supported.
+This filter Used differently by `allow`/`deny` handlers. Allow handler pass query only if all tables specified in configuration file matched with tables in the query and deny if at least one specified table not matched.
+Deny handler denies query if at least one table specified in the configuration file matches tables in the query. Look at examples below.
+
+### Allow handler
+
+Allow handler denies any query if it does not contain the whole set of tables specified in configuration file for this handler. Query is allowed if number of tables in configuration file equals to number of tables in the query. E.g. all tables from configuration file should be presented in a set of tables of the query.
```

Comment:
- Formatting, wording, grammar.
- Changed the wording to not imply that query *must* use all tables listed (that is, not using a table from the list does not deny the query).

```suggestion
`allow` handler allows queries if they contain only tables specified in the configuration file.
If a query uses a table not from the provided list, `allow` handler does not apply
and the following handlers will be considered next.
```

Reply:
actually, it will deny query that uses tables not from the list. and query **must** use all tables in current implementation.
changed.